data-prepper-api/src/main/java/org/opensearch/dataprepper/model/annotations/ExampleValues.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.model.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Use this annotation to provide example values for plugin configuration.
+ *
+ * @since 2.11
+ */
+@Documented
+@Retention(RUNTIME)
+@Target({FIELD})
+public @interface ExampleValues {
+    /**
+     * One or more examples.
+     * @return the examples.
+     * @since 2.11
+     */
+    Example[] value();
+
+    /**
+     * A single example.
+     *
+     * @since 2.11
+     */
+    @interface Example {
```

Comment:
Does it have to be an inner interface? As it is inner, at the time of specifying these example values it looks overly verbose 
```
@ExampleValues({
                @ExampleValues.Example("some example value"),
                @ExampleValues.Example(value = "second example value", description = "This is the second value.")
        })
```

Reply:
This is actually quite common for annotations that are coupled with their parent. You can add `ExampleValues` as a static import.

Reply:
I pushed a revision that imports `ExampleValues.Example`.